src/System.Net.Http/src/System/Net/Http/OSX/CurlHandler.SslProvider.cs

```diff
@@ -58,24 +71,23 @@ internal static void SetSslOptions(EasyRequest easy, ClientCertificateOption cli
                     // user's keychain and setting the SSL policy trust for it to "Always Trust".
                     // Similarly, the "block this" could be attained by setting the SSL policy for a cert in the
                     // keychain to "Never Trust".
-                    throw new PlatformNotSupportedException(
-                        SR.Format(
-                            SR.net_http_libcurl_callback_notsupported,
-                            CurlVersionDescription,
-                            CurlSslVersionDescription));
-                }
-
-                // Revocation checking is always on for darwinssl (SecureTransport).
-                // If any other backend is used and revocation is requested, we can't guarantee
-                // that assertion.
-                if (easy._handler.CheckCertificateRevocationList &&
-                    !CurlSslVersionDescription.Equals("SecureTransport"))
-                {
-                    throw new PlatformNotSupportedException(
-                    SR.Format(
-                        SR.net_http_libcurl_revocation_notsupported,
-                        CurlVersionDescription,
-                        CurlSslVersionDescription));
+                    //
+                    // However, one case we can support is when we know all certificates will pass validation.
+                    // We can detect a key case of that: whether DangerousAcceptAnyServerCertificateValidator was used.
+                    if (easy.ServerCertificateValidationCallbackAcceptsAll)
+                    {
+                        EventSourceTrace("Disabling peer and host verification per {0}", nameof(HttpClientHandler.DangerousAcceptAnyServerCertificateValidator), easy: easy);
+                        easy.SetCurlOption(Interop.Http.CURLoption.CURLOPT_SSL_VERIFYPEER, 0); // don't verify the peer
+                        easy.SetCurlOption(Interop.Http.CURLoption.CURLOPT_SSL_VERIFYHOST, 0); // don't verify the peer cert's hostname
```

Comment:
darwinssl, at least, uses VERIFYHOST=1 as "should I enable SNI".  So it would be good if we can leave this on.  But maybe that causes it to still reject the self-signed cert if the hostname doesn't match, in which case... cutting SNI is a sad side effect that we may need to write down.  (Though I don't know what the state of SNI is for us on Linux or Windows... so maybe this is  "a nice feature of macOS might get turned off"))